### PR TITLE
Issue 240 - Add completed calls list to dashboard view

### PR DESCRIPTION
**Pull request by charleshuang80**
I rule and have completed some work on Case Manager that's ready for review!

changes in commit for Issue #240
* create completed calls partial
* add the partial to dashboard view

changes in commits for issues #214:
* collapse/change all partials for dashboards index and search views into _table_content partial.
* change partials call for add and remove pregnancy js erb files to pass tests (but not sure where these actions are being used)

### CODE REVIEW

File: app/views/dashboards/index.html.erb [hunk 1]
```diff
@@ -2,8 +2,9 @@
   <div class="col-md-12">
     <%= render partial: 'dashboards/overview' %>
     <%= render partial: 'dashboards/search_form' %>
-    <%= render partial: 'dashboards/call_list', locals: { pregnancies: current_user.pregnancies } %>
-    <%= render partial: 'dashboards/urgent_pregnancies', locals: { pregnancies: @urgent_pregnancies } %>
+    <%= render partial: 'dashboards/table_content', locals: { title: 'Your call list', table_type: 'call_list', pregnancies: current_user.pregnancies } %>
```
Comment (colinxfleming) [inline app/views/dashboards/index.html.erb]: Yesssss

File: app/views/dashboards/search.js.erb [hunk 2]
```diff
@@ -1,2 +1,2 @@
-$('#search_results').empty(); 
-$("#search_results").append("<%= escape_javascript(render partial: 'dashboards/search_results', locals: { pregnancies: @results } ) %>");
+$('#search_results').empty();
+$("#search_results").append("<%= escape_javascript(render partial: 'dashboards/table_content', locals: { title: 'Search results', table_type: 'search_results', pregnancies: @results } ) %>");
```
Comment (colinxfleming) [inline app/views/dashboards/search.js.erb]: 💯

File: app/views/users/remove_pregnancy.js.erb [hunk 3]
```diff
@@ -1,2 +1,3 @@
-$('#call_list').empty(); 
-$('#call_list').append('<%= escape_javascript(render partial: "dashboards/call_list", locals: { pregnancies: @user.pregnancies }) %>')
+$('#call_list').empty();
+$('#call_list').append('<%= escape_javascript(render partial: "dashboards/table_content", locals: { title: 'call list', table_type: 'call_list', pregnancies: @user.pregnancies }) %>')
+// if this table is on the dashboards#index page then title should be edited and @user change to current_user
```
Comment (colinxfleming) [inline app/views/users/remove_pregnancy.js.erb]: Agree! If `current_user` is accessible from these actions, let's definitely use that instead of `@user`.

**Comment by colinxfleming** [comment]
This is really excellent work, @charleshuang80 ! Will make editing the dashboards as we tighten the bolts so much easier. Thank you!

Appreciated your note about `@user` instead of `current_user` in the javascripts, as well. If you have the opportunity to close then feel free to try it out and make sure it works; otherwise I'll make an issue.

I'll do a quick check on Friday and then merge.

File: app/views/users/remove_pregnancy.js.erb (hunk 3 above)
Comment (colinxfleming) [inline app/views/users/remove_pregnancy.js.erb]: PS your guess is right, these trigger from dashboards#index.

**Comment by colinxfleming** [comment]
Also 💯 for issue linking and PR template!

**Comment by charleshuang80** [comment]
I'd prefer to make the changing the `@user` to `current_user` in the js erb files in another ticket. I hadn't figured out where to trigger those views, so wouldn't be able to test them very quickly. I can take care of that issue, but don't want to hold up the merge.

**Comment by colinxfleming** [comment]
Post a screenshot of how this renders in practice for posterity:

screen shot 2016-04-23 at 2 55 11 pm

**Comment by colinxfleming** [comment]
It looks good to me and does what it says on the label! Thanks for pushing this thru @charleshuang80 !

**Pull request merged**